onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -107,6 +111,276 @@ std::shared_ptr<KernelRegistry> GetMklDnnKernelRegistry() {
+                input_from_subgraph = false;
+              }
+            }
+            if (input_from_subgraph == false) {
+              CreateMetaDef(sub_var, subgraph_attributes, result);
+              subgraph_attributes.clear();
+              output_to_source_node_map.clear();
+            }
+          }
+        }
+        if (!sub_var.subgraph_node_indexes.empty()) {
+          if (static_cast<int>(node->GetOutputEdgesCount()) > 1) {
+            // If current node has branches
+            //    iterate and see if all nodes are mkldnn ops OR
+            //      it ends in node with same number of input edges (mkldnn node or cpu node)
+            //      create sub-graph
+            bool create_subgraph = false;
+            bool break_loop = false;
+            while (!break_loop) {
+              if (temp_index > graph_viewer.MaxNodeIndex())
+                break_loop = true;
+
+              auto next_node = graph_viewer.GetNode(temp_index);
+              if (next_node->GetInputEdgesCount() == node->GetOutputEdgesCount()) {
+                // if all nodes in the branch loop are mkldnn nodes
+                // then continue with adding nodes to sub-graph
+                break_loop = true;
+              }
+              // inner nodes. if inner nodes are not  mkldnn nodes
+              // create subgraph (inception v2)
+              std::vector<std::string>::iterator sub_it = std::find(
+                  mkl_ops.begin(), mkl_ops.end(), next_node->OpType());
+              if (sub_it == mkl_ops.end()) {
+                // break and create a sub-graph
+                break_loop = true;
+                create_subgraph = true;
+              }
+              temp_index++;
+            }
+            if (create_subgraph) {
+              CreateMetaDef(sub_var, subgraph_attributes, result);
+              subgraph_attributes.clear();
+              output_to_source_node_map.clear();
+            }
+          }
+        }
+      }
+    } else {
+      if (!sub_var.subgraph_node_indexes.empty()) {
+        CreateMetaDef(sub_var, subgraph_attributes, result);
+        subgraph_attributes.clear();
+      }
+    }
+    node_index++;
+  }  // graph_viewer node iterator ends
+  if (!sub_var.subgraph_node_indexes.empty()) {
+    CreateMetaDef(sub_var, subgraph_attributes, result);
+    subgraph_attributes.clear();
+  }
+  return result;
+}
+
+void MKLDNNExecutionProvider::CreateMetaDef(SubgraphVariables& sub_var, const NodeAttributes& subgraph_attributes,
+                                            std::vector<std::unique_ptr<ComputeCapability>>& result) const {
+  std::string graph_fused_nodes;
+  std::string node_list;
+  std::string subgraph_id = std::to_string(sub_var.subgraph_index);
+  sub_var.subgraph_index++;
+
+  auto meta_def = std::make_unique<::onnxruntime::IndexedSubGraph::MetaDef>();
+  meta_def->name = "MkldnnCustomOp" + std::to_string(sub_var.subgraph_index);
+  meta_def->domain = kMklDnnExecutionProvider;
```

Comment:
this should be domain of the op implementation. put kMSDomain for now (meaning custom implementation)

Reply:
changed to kMSDomain.